llvm/utils/TableGen/AsmMatcherEmitter.cpp

```diff
@@ -2019,7 +2058,7 @@ emitConvertFuncs(CodeGenTarget &Target, StringRef ClassName,
   CvtOS
       << "                              std::begin(TiedAsmOperandTable)) &&\n";
   CvtOS << "             \"Tied operand not found\");\n";
-  CvtOS << "      unsigned TiedResOpnd = TiedAsmOperandTable[OpIdx][0];\n";
+  CvtOS << "      unsigned TiedResOpnd = TiedAsmOperandTable[*(p + 1)][0];\n";
```

Comment:
Please check if #81381 resolves this issue.

Reply:
It does not, it breaks the tied operands resolution for me, I run into the same issue I had before of the wrong value being used. I think `*(p + 1)` is correct as offsets should not be applied here?

Reply:
> I think *(p + 1) is correct as offsets should not be applied here?

Agreed. It looks like it is not an operand index at all.

Reply:
I guess the assert above should also be fixed.
And we shouldn't be calculating OpIdx before the switch because it can lead to OOB access...

Reply:
Fixed the assert. Moved this bug fix to a separate patch (https://github.com/llvm/llvm-project/pull/83588).

The OpIdx calculation is needed for the generated switch cases below so I didnt remove it for now. Im not certain how I would remove it just for this case except for adding a if statement? I cant imagine a circumstance where it generates an OOB access though?

Reply:
> I cant imagine a circumstance where it generates an OOB access though?

OOB access can happen in HasOptionalOperands case if `*(p + 1)` (index into TiedAsmOperandTable) is larger than the size of DefaultsOffset array.